[CI] SpatialCentroidTests testAggregateIntermediate {TestCase=<geo_point> #4} failing

**Build Scans:**
- [elasticsearch-periodic #8062 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/nsd43n36wyaxw)
- [elasticsearch-periodic #8056 / openjdk21_checkpart3_java-matrix](https://gradle-enterprise.elastic.co/s/ur2itr5ufvnka)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests.testAggregateIntermediate {TestCase=<geo_point> #4}" -Dtests.seed=593FD7B5E38466D9 -Dtests.locale=pt-TL -Dtests.timezone=America/Adak -Druntime.java=23
```

**Applicable branches:**
8.17

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testAggregateIntermediate%20%7BTestCase%3D%3Cgeo_point%3E%20%234%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <[1 1 0 0 0 d 8 7d 67 57 2c 2 c0 30 50 e7 29 7c 8c 11 c0]>
     but: was <[1 1 0 0 0 c 8 7d 67 57 2c 2 c0 30 50 e7 29 7c 8c 11 c0]>
```

**Issue Reasons:**
- [8.17] 2 failures in test testAggregateIntermediate {TestCase=<geo_point> #4} (2.1% fail rate in 97 executions)
- [8.17] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -52,12 +52,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.MlJobIT
   method: testMultiIndexDelete
   issue: https://github.com/elastic/elasticsearch/issues/112381
-- class: org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests
-  method: "testAggregateIntermediate {TestCase=<geo_point> #2}"
-  issue: https://github.com/elastic/elasticsearch/issues/112461
-- class: org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests
-  method: testAggregateIntermediate {TestCase=<geo_point>}
-  issue: https://github.com/elastic/elasticsearch/issues/112463
 - class: org.elasticsearch.xpack.inference.external.http.RequestBasedTaskRunnerTests
   method: testLoopOneAtATime
   issue: https://github.com/elastic/elasticsearch/issues/112471

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialCentroidTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialCentroidTests.java
@@ -23,13 +23,15 @@ import org.elasticsearch.xpack.esql.expression.function.FunctionName;
 import org.elasticsearch.xpack.esql.expression.function.MultiRowTestCaseSupplier;
 import org.elasticsearch.xpack.esql.expression.function.MultiRowTestCaseSupplier.IncludingAltitude;
 import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
 
-import java.nio.ByteOrder;
 import java.util.List;
 import java.util.function.Supplier;
 import java.util.stream.Stream;
 
-import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.closeTo;
 
 @FunctionName("st_centroid_agg")
 public class SpatialCentroidTests extends AbstractAggregationTestCase {
@@ -74,16 +76,58 @@ public class SpatialCentroidTests extends AbstractAggregationTestCase {
                 count++;
             }
 
-            var expected = new BytesRef(
-                WellKnownBinary.toWKB(new Point(xSum.value() / count, ySum.value() / count), ByteOrder.LITTLE_ENDIAN)
-            );
+            var expectedX = xSum.value() / count;
+            var expectedY = ySum.value() / count;
 
             return new TestCaseSupplier.TestCase(
                 List.of(fieldTypedData),
                 "SpatialCentroid[field=Attribute[channel=0]]",
                 fieldTypedData.type(),
-                equalTo(expected)
+                centroidMatches(expectedX, expectedY, 1e-14)
             );
         });
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private static Matcher<BytesRef> centroidMatches(double x, double y, double error) {
+        return new TestCentroidMatcher(x, y, error);
+    }
+
+    private static class TestCentroidMatcher extends BaseMatcher<BytesRef> {
+        private final double x;
+        private final double y;
+        private final Matcher<Double> mx;
+        private final Matcher<Double> my;
+
+        private TestCentroidMatcher(double x, double y, double error) {
+            this.x = x;
+            this.y = y;
+            this.mx = closeTo(x, error);
+            this.my = closeTo(y, error);
+        }
+
+        @Override
+        public boolean matches(Object item) {
+            if (item instanceof BytesRef wkb) {
+                var point = (Point) WellKnownBinary.fromWKB(GeometryValidator.NOOP, false, wkb.bytes, wkb.offset, wkb.length);
+                return mx.matches(point.getX()) && my.matches(point.getY());
+            }
+            return false;
+        }
+
+        @Override
+        public void describeMismatch(Object item, Description description) {
+            if (item instanceof BytesRef wkb) {
+                var point = (Point) WellKnownBinary.fromWKB(GeometryValidator.NOOP, false, wkb.bytes, wkb.offset, wkb.length);
+                description.appendText("was ").appendValue(point);
+            } else {
+                description.appendText("was ").appendValue(item);
+            }
+        }
+
+        @Override
+        public void describeTo(Description description) {
+            description.appendValue("    POINT (" + x + " " + y + ")");
+        }
     }
 }

